nexus_constructor/main_window.py

```diff
@@ -27,11 +29,22 @@
     LinkTransformation,
 )
 from nexus_constructor.json import filewriter_json_writer
+from nexus_constructor.json.filewriter_json_reader import json_to_nexus
 
 NEXUS_FILE_TYPES = {"NeXus Files": ["nxs", "nex", "nx5"]}
 JSON_FILE_TYPES = {"JSON Files": ["json", "JSON"]}
 
 
+def show_warning_dialog(message: str, title: str, additional_info: Optional[str]):
```

Comment:
I think we should put this in ui_utils, as there is at least 1 more place that uses a messagebox in the same way.